.circleci/config.yml

```diff
@@ -1,68 +1,84 @@
-            | jq -r ".base.sha | select(. != null)")...
-        else
-          BASE_COMMIT=HEAD^
-        fi
-        if [[ $BASE_COMMIT ]]; then
-          CHANGED=$(git diff --name-only $BASE_COMMIT)
-          echo "$CHANGED"
-          echo "export CHANGED=\"$CHANGED\"" >> $BASH_ENV
-        fi
+version: 2.1
 
-node_modules_cache_key: &node_modules_cache_key
-  yarn-deps-v2-{{ checksum "yarn.lock" }}
-
-restore_node_modules_cache: &restore_node_modules_cache
-  restore_cache:
-    keys:
-      - *node_modules_cache_key
-      - yarn-deps-v2-
+commands:
+  checkout_repo:
+    steps:
+      - checkout
+      - run:
+          name: Checkout submodule
+          command: |
+              git submodule sync
+              git submodule update --init --recursive
+      - run:
+          name: Get changed files
+          command: |
+              if [[ $CIRCLE_PULL_REQUEST ]]; then
+                BASE_COMMIT=$(curl -s -H "Authorization: token a61ecb2fc5b72da54431""1b3db3875c96854958a8" \
+                  https://api.github.com/repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/pulls/$CIRCLE_PR_NUMBER \
+                  | jq -r ".base.sha | select(. != null)")...
+              else
+                BASE_COMMIT=HEAD^
+              fi
+              if [[ $BASE_COMMIT ]]; then
+                CHANGED=$(git diff --name-only $BASE_COMMIT)
+                echo "$CHANGED"
+                echo "export CHANGED=\"$CHANGED\"" >> $BASH_ENV
+              fi
 
-yarn_install: &yarn_install
-  run:
-    name: Install dependencies
-    command: yarn
+  skip_if_only_changed:
+    parameters:
+      filter:
+        type: string
+    steps:
+      - run:
+          name: Skip tests if only "<< parameters.filter >>" are changed
+          command: echo "$CHANGED" | grep -qvE '<< parameters.filter >>' || circleci step halt
 
-codecov: &codecov
-  run:
-    name: Upload code coverage reports to Codecov
-    command: ./node_modules/.bin/codecov
+  install_dependencies:
+    steps:
+      - restore_cache:
+          keys:
+            - yarn-deps-v2-{{ checksum "yarn.lock" }}
+            - yarn-deps-v2-
+      - run:
+          name: Install dependencies
+          # upgrade Yarn before installing dependencies, if needed
+          command: yarn check || { curl -o- -L https://yarnpkg.com/install.sh | bash && yarn; }
+      - save_cache:
+          key: yarn-deps-v2-{{ checksum "yarn.lock" }}
+          paths:
+            - node_modules
 
-screenshotter: &screenshotter
-  steps:
-    - checkout
-    - *post_checkout
-    - run:
-        name: Skip screenshotter if no KaTeX code has been changed
-        command: |
-            if [[ $CHANGED ]]; then
-              echo "$CHANGED" | grep -qvE '^contrib/|^docs/|^static/|^website/|^LICENSE|\.md$' || circleci step halt
-            fi
+  codecov:
+    parameters:
+      flag:
+        type: string
```

Comment:
I didn't realize commands could be parameterized.  This is super powerful.

Reply:
@kevinbarabash Custom commands and parameters are new features in CircleCI 2.1.